File: HorseKata/java/src/main/java/PaginatedTable.java
import java.util.List;
import java.util.Objects;

public class PaginatedTable {
    final List<String> headers;
    final List<List<String>> tableData;
    final int totalRows;

    public PaginatedTable(List<String> headers, List<List<String>> tableData, int totalRows) {
        this.headers = headers;
        this.tableData = tableData;
        this.totalRows = totalRows;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        PaginatedTable that = (PaginatedTable) o;
        return totalRows == that.totalRows &&
                Objects.equals(headers, that.headers) &&
                Objects.equals(tableData, that.tableData);
    }

    @Override
    public int hashCode() {
        return Objects.hash(headers, tableData, totalRows);
    }
}

File: HorseKata/csharpcore/Horse/PaginatedTable.cs
using System;
using System.Collections.Generic;

namespace HorseKata
{
    /**
 * instances of this class can be returned to the front end for display
 */
    public class PaginatedTable
    {
        /**
        * the headers for the table
        */
        public List<string> Headers { get; }

        /**
         * The data to show in the current page of the table
         */
        public List<List<string>> TableData { get; }

        /**
         * the total number of rows, including those rows not shown on the current page
         */
        public int TotalRows { get; }

        public PaginatedTable(List<string> headers, List<List<string>> tableData, int totalRows) 
        {
            Headers = headers;
            TableData = tableData;
            TotalRows = totalRows;
        }

        public override bool Equals(object o) 
        {
            if (this == o) return true;
            if (o == null || GetType() != o.GetType()) return false;
            PaginatedTable that = (PaginatedTable) o;
            return TotalRows == that.TotalRows &&
                   Equals(Headers, that.Headers) &&
                   Equals(TableData, that.TableData);
        }

        public override int GetHashCode()
        {
            return Tuple.Create(Headers, TableData, TotalRows).GetHashCode();
        }
    }
}